cmd/gapit/replace_resource.go

```diff
@@ -90,36 +94,100 @@ func (verb *replaceResourceVerb) Run(ctx context.Context, flags flag.FlagSet) er
+
+	switch {
+	case verb.Handle != "":
+		matchedResource, err := resources.FindSingle(func(t api.ResourceType, r service.Resource) bool {
+			return strings.Contains(r.GetHandle(), verb.Handle)
+		})
+		if err != nil {
+			return err
+		}
+		resourcePath = capture.Command(uint64(verb.At)).ResourceAfter(matchedResource.ID).Path()
+		newResourceBytes, err := ioutil.ReadFile(verb.ResourcePath)
+		if err != nil {
+			return log.Errf(ctx, err, "Could not read resource file %s", verb.ResourcePath)
+		}
+		resourceData = api.NewResourceData(&api.Shader{Type: api.ShaderType_Spirv, Source: string(newResourceBytes)})
+	case verb.UpdateResourceBinary != "":
+		shaderResources := resources.FindAll(func(t api.ResourceType, r service.Resource) bool {
+			return t == api.ResourceType_ShaderResource
+		})
+		ids := make([]*path.ID, len(shaderResources))
+		resourcesSource := make([]*api.ResourceData, len(shaderResources))
+		for i, v := range shaderResources {
+			ids[i] = v.ID
+			resourcePath := capture.Command(uint64(verb.At)).ResourceAfter(v.ID)
+			rd, err := client.Get(ctx, resourcePath.Path())
 			if err != nil {
-				return log.Errf(ctx, err, "Could not read resource file %s", verb.ResourcePath)
+				log.Errf(ctx, err, "Could not get data for shader: %v", v)
+				return err
 			}
-
-			newResourceData := api.NewResourceData(&api.Shader{Type: api.ShaderType_SpirvBinary, Source: string(newResourceBytes)})
-			newResourcePath, err := client.Set(ctx, resourcePath.Path(), newResourceData)
+			newData, err := verb.getNewResourceData(ctx, rd.(*api.ResourceData).GetShader().GetSource())
 			if err != nil {
-				return log.Errf(ctx, err, "Could not update data for shader: %v", matchedResource)
+				log.Errf(ctx, err, "Could not update the shader: %v", v)
+				return err
 			}
-			newCapture := newResourcePath.GetResourceData().GetAfter().GetCapture()
-			newCaptureFilepath, err := filepath.Abs(verb.OutputTraceFile)
-			err = client.SaveCapture(ctx, newCapture, newCaptureFilepath)
+			resourcesSource[i] = api.NewResourceData(&api.Shader{Type: api.ShaderType_Spirv, Source: string(newData)})
+		}
+		resourceData = api.NewMultiResourceData(resourcesSource)
+		resourcePath = capture.Command(uint64(verb.At)).ResourcesAfter(ids).Path()
+	}
+
+	newResourcePath, err := client.Set(ctx, resourcePath, resourceData)
+	if err != nil {
+		return log.Errf(ctx, err, "Could not update resource data: %v", resourcePath)
+	}
+	newCapture := path.FindCapture(newResourcePath.Node())
+	newCaptureFilepath, err := filepath.Abs(verb.OutputTraceFile)
+	err = client.SaveCapture(ctx, newCapture, newCaptureFilepath)
+
+	log.I(ctx, "Capture written to: %v", newCaptureFilepath)
+	return nil
+}
+
+// getNewResourceData runs the update resource binary on the old resource data
+// and returns the newly generated resource data
+func (verb *replaceResourceVerb) getNewResourceData(ctx context.Context, resourceData string) (string, error) {
+	updateCmd := exec.Command(verb.UpdateResourceBinary)
```

Comment:
Did my example code not work for you?

Reply:
It does. Sorry, totally missed that comment.